app/controllers/projects/pledges_controller.rb

```diff
@@ -6,12 +6,13 @@ def new
 
   def create
     @pledge = Pledge.create(pledge_params)
-    if @pledge.save
+    if @pledge.save && current_user.authenticate(params[:password])
```

Comment:
just as a btw: this will probably break sometime tomorrow as we need to drop the passwords from params.